### PR TITLE
Update code for pytorch 1.9.0

### PR DESCRIPTION
**Pull request by wanchichen**
The [latest version of pytorch](https://github.com/pytorch/pytorch/commit/58eb23378f2a376565a66ac32c93a316c45b6131#diff-b3c160475f0fbe8ad50310f92d3534172ba98203387a962b7dc8f4a23b15cf4dL35) removed some internal code that this repo imports, such as:

 - `container_abcs` from `torch._six`
 - `int_classes` from `torch._six`

Causing anyone running the current TimeSformer code on new versions of pytorch to receive compile-time errors.
The fix is similar to https://github.com/NVIDIA/apex/pull/1049, checking the user's pytorch version before importing the necessary components.

### CODE REVIEW

**Comment by smivv** [comment]
Hello, sorry for bothering, but why this PR still wasn't merged?

**Comment by wanchichen** [comment]
The original author of the paper/repo (@gberta) no longer works at Facebook, so he doesn’t have permissions to merge it. So unless someone there with the necessary permissions takes notice it likely won’t be merged for the foreseeable future.

**Comment by smivv** [comment]
@wanchichen, thank you for your answer!

By the way, does your branch works fine? 😄 
I consider your PR as the most recent update, so probably will use it.

**Comment by wanchichen** [comment]
Yeah the branch should work without issues.

**Comment by ShoufaChen** [comment]
Hi, @wanchichen 
Thanks for your sharing.

One small fix:
you should add 
```
import torch
``` 
at the top of `timesformer/datasets/multigrid_helper.py`.

**Comment by wanchichen** [comment]
Thank you @ShoufaChen , it has been fixed.

**Comment by MeSveta** [comment]
hi, I'm trying to set the environment but without any success. How I should update the pull request? manually? since it is not merged.

**Comment by wanchichen** [comment]
You can just clone from my fork: `git clone https://github.com/wanchichen/TimeSformer.git`

**Comment by MeSveta** [comment]
Hello William,
Many thanks!!
Can I please ask you another question? I'm more interested in working with
the HowTo100 dataset , I'm trying to understand where to get the
translation from a label that is a number to the actual class of the video
( such as 'preparing breakfast').
Any ideas?

Thanks:)

**Comment by jromerooo2** [comment]
@MeSveta From my understanding you should create your own function that maps those features to an integer and/or viceversa. There should be a file inside of the dataset that contains the video path or video as an array and labels (sometimes the labels are in the actual name of the video, this depends on the structure of the dataset).

An example for the described above could be:
`class_labels = sorted({str(path).split("/")[2] for path in all_video_file_paths})`
`label2id = {label: i for i, label in enumerate(class_labels)}`
`id2label = {i: label for label, i in label2id.items()}`

This splits the path of the file in the current index `i` of the iteration and assigns it to a set. The set can't contain any duplicates so that is already covered for us.

Note that the code heavily relies on the assumption that your dataset is structured as the following:
'subset/train/Class/class_1.mp4'

So please modify if your dataset looks different.
